### PR TITLE
Implement Spider

### PR DESCRIPTION
**Pull request by CortexPE**

### CODE REVIEW

File: Horion/Module/Modules/Spider.cpp [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+const char* Spider::getModuleName() {
+	return "Spider";
+}
+
+void Spider::onMove(C_MoveInputHandler* input) {
+	if (g_Data.getLocalPlayer() == nullptr)
+		return;
+	C_LocalPlayer* player = g_Data.getLocalPlayer();
+
+	if (player->isInLava() == 1 || player->isInWater() == 1)
```
Comment (GodGamer029) [inline Horion/Module/Modules/Spider.cpp]: ?

File: Horion/Module/Modules/Spider.cpp [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+	
+	static std::vector<vec3_ti*> sideBlocks;
+	if (sideBlocks.empty()) {
+		sideBlocks.push_back(new vec3_ti(0, 0, -1));
+		sideBlocks.push_back(new vec3_ti(0, 0, 1));
+		sideBlocks.push_back(new vec3_ti(-1, 0, 0));
+		sideBlocks.push_back(new vec3_ti(1, 0, 0));
+	}
+
+	for (auto current : sideBlocks) {
```
Comment (GodGamer029) [inline Horion/Module/Modules/Spider.cpp:36]: a nested for-loop would have been more appropriate here imo

Comment (GodGamer029) [reply Horion/Module/Modules/Spider.cpp:36]: if im not mistaken there even is a boolean inside the player struct that is true when the player collides with walls horizontally

**Review by GodGamer029 — CHANGES_REQUESTED** [review]

File: Horion/Module/Modules/Spider.cpp (hunk 2 above)
Comment (CortexPE) [reply Horion/Module/Modules/Spider.cpp:36]: Not sure what property you're talking about... I tried to look for it before coming up with the side-block check

File: Horion/Module/Modules/Spider.cpp (hunk 1 above)
Comment (richardletshacks) [reply Horion/Module/Modules/Spider.cpp]: the `` == 1`` is rendundant @CortexPE

**Pull request merged**